Services/Facades/BetFacade.cs

```diff
@@ -13,57 +13,86 @@ public class BetFacade : IBetFacade
             _betQuoteService = betQuote;
+            _dbContext = dbContext;
         }
 
         public async Task<BetRequestDto> CreateBetAsync(CreateBetRequestDto newFullBet)
         {
             CreateBetQuotesDto newBetQuote = _mapper.Map<CreateBetQuotesDto>(newFullBet);
             CreateBetsDto newBets = _mapper.Map<CreateBetsDto>(newFullBet);
-            BetsDto createdBet = await _betService.CreateAsync(newBets);
-            if (createdBet == null)
+
+            using var transaction = _dbContext.Database.BeginTransaction();
+            try
             {
-                return null;
-            }
+                BetsDto createdBet = await _betService.CreateAsync(newBets);
+                if (createdBet == null)
+                {
+                    transaction.Rollback();
+                    return null;
+                }
 
-            BetQuoteDto createdBetQuote = await _betQuoteService.CreateAsync(newBetQuote, createdBet.Id);
-            if (createdBetQuote == null)
+                BetQuoteDto createdBetQuote = await _betQuoteService.CreateAsync(newBetQuote, createdBet.Id);
+                if (createdBetQuote == null)
+                {
+                    transaction.Rollback();
+                    return null;
+                }
+                transaction.Commit();
+                return new BetRequestDto { bet = createdBet, betQuote = createdBetQuote };
+            }
+            catch (Exception ex)
             {
-                return null;
+                transaction.Rollback();
+                throw ex;
             }
-
-            return new BetRequestDto { bet = createdBet, betQuote = createdBetQuote };
         }
 
         public async Task<BetRequestDto> UpdateBetAsync(Guid betId, UpdateBetRequestDto updatedFullBet)
         {
             UpdateBetQuotesDto updatedBetQuote = _mapper.Map<UpdateBetQuotesDto>(updatedFullBet);
             UpdateBetsDto updatedBets = _mapper.Map<UpdateBetsDto>(updatedFullBet);
-            BetQuoteDto currentBetQuote = await _betQuoteService.GetByBetIdAsync(betId);
-            if (currentBetQuote == null)
-            {
-                return null;
-            }
 
-            BetQuoteDto newBetQuote = await _betQuoteService.UpdateById(currentBetQuote.Id, updatedBetQuote);
-            if (newBetQuote == null)
+            using var transaction = _dbContext.Database.BeginTransaction();
+            try
             {
-                return null;
-            }
+                BetQuoteDto currentBetQuote = await _betQuoteService.GetByBetIdAsync(betId);
+                if (currentBetQuote == null)
+                {
+                    transaction.Rollback();
+                    return null;
+                }
+
+                BetQuoteDto newBetQuote = await _betQuoteService.UpdateById(currentBetQuote.Id, updatedBetQuote);
+                if (newBetQuote == null)
+                {
+                    transaction.Rollback();
```

Comment:
no need for rollbacks if the _betQuoteService.UpdateById will throw an exception if the entity is null - which I suppose it happens in the other PR